src/realm/object-store/sync/app.hpp

```diff
@@ -267,6 +267,25 @@ class App : public std::enable_shared_from_this<App>,
     static SharedApp get_cached_app(const std::string& app_id,
                                     const std::optional<std::string>& base_url = std::nullopt);
 
+    /// Get the current base URL for the AppServices server used for http requests and sync
+    /// connections.
+    /// If an update_base_url() operation is currently in progress, this value will not be
+    /// updated with the new value until that operation is complete.
+    /// @return String containing the current base url value
+    std::string get_base_url() const;
+
+    /// Update the base URL after the app has been created. The location info will be retrieved
+    /// using the provided base URL. If this operation fails, the app will continue to use the original base URL and
+    /// false will be returned. If the operation is successful, the app and sync client will use the new location info
```

Comment:
there is no return type. you probably meant the completion callback is invoked with an error.